indico/modules/events/surveys/clone.py

```diff
@@ -0,0 +1,56 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2021 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from indico.core.db import db
+from indico.core.db.sqlalchemy.util.models import get_simple_column_attrs
+from indico.modules.events.cloning import EventCloner
+from indico.modules.events.surveys.models.surveys import Survey
+from indico.util.i18n import _
+
+
+class EventSurveyCloner(EventCloner):
+    name = 'event_survey'
+    friendly_name = _('Surveys')
+    is_default = False
+
+    @property
+    def is_available(self):
+        return self._has_content(self.old_event)
+
+    def has_conflicts(self, target_event):
+        return self._has_content(target_event)
+
+    def run(self, new_event, cloners, shared_data, event_exists=False):
+        with db.session.no_autoflush:
+            self._clone_surveys(new_event)
+        db.session.flush()
+
+    def _has_content(self, event):
+        return len([survey for survey in event.surveys if not survey.is_deleted])
+
+    def _clone_surveys(self, new_event):
+        survey_attrs = get_simple_column_attrs(Survey) - {'uuid', 'start_dt', 'end_dt', '_last_friendly_submission_id'}
+        for old_survey in self.old_event.surveys:
```

Comment:
`Event.surveys` doesn't exclude deleted surveys, so we need to skip those explicitly

```suggestion
        for old_survey in self.old_event.surveys:
            if old_survey.is_deleted:
                continue
```